plugins/textwatcher/plugin.js

```diff
@@ -253,7 +232,18 @@
 				return;
 			}
 
-			this._buffer.input( selectionRange );
+			var matched = this.callback( selectionRange );
```

Comment:
I pushed solution based on above in the [t/2373b](https://github.com/ckeditor/ckeditor-dev/tree/t/2373b) branch. It also results with smaller LoC added compared to the original code.

Reply:
Note that there's one issue - in case if during thorttled call there's no selection (edge case, but we need to consider it) there's an early return.

https://github.com/ckeditor/ckeditor-dev/blob/b5e9817/plugins/textwatcher/plugin.js#L177-L179

It would be much better to notify unmatch, because we're no longer matching anything, right? It would be nice to add unit test coverage for this edge case.